lib/internal/modules/cjs/loader.js

```diff
@@ -522,71 +511,24 @@ Module._resolveLookupPaths = function(request, parent, newReturn) {
     }
 
     debug('looking for %j in %j', request, paths);
-    return (newReturn ? (paths.length > 0 ? paths : null) : [request, paths]);
+    return paths.length > 0 ? paths : null;
   }
 
   // with --eval, parent.id is not set and parent.filename is null
   if (!parent || !parent.id || !parent.filename) {
     // Make require('./path/to/foo') work - normally the path is taken
     // from realpath(__filename) but with eval there is no filename
-    var mainPaths = ['.'].concat(Module._nodeModulePaths('.'), modulePaths);
+    const mainPaths = ['.'].concat(Module._nodeModulePaths('.'), modulePaths);
 
     debug('looking for %j in %j', request, mainPaths);
-    return (newReturn ? mainPaths : [request, mainPaths]);
-  }
-
-  // Is the parent an index module?
-  // We can assume the parent has a valid extension,
-  // as it already has been accepted as a module.
-  const base = path.basename(parent.filename);
-  var parentIdPath;
-  if (base.length > indexLen) {
```

Comment:
Can you explain why this conditional is also dead code? I'm not familiar with this check at all.

Reply:
This code path is long unused. In core code we only return `parentDir` from here on but this code part is about `id`. It was originally added to fix child modules getting a wrong id value (https://github.com/nodejs/node/commit/6301f18e62e8f90026dda3aec3457e4723ea319b). This function will not set the child's id and therefore it is dead code.

Reply:
Ahh, so the code runs, its side-effect is just setting `id` which is itself no longer used? Thanks, missed that!

Reply:
Yes, exactly :)